reckless_github/src/repository.rs

```diff
@@ -30,6 +32,121 @@ impl Github {
             plugins: vec![],
         }
     }
+
+    /// Index the repository to store information
+    /// related to the plugins
+    pub fn index_repository(&mut self) {
+        let mut list_plugins = Vec::<Plugin>::new();
+        let repo_path = get_dir_path_from_url(&self.url);
+        let pattern = format!("{}/[!.]*/*", &repo_path);
+        for plugin in glob(&pattern).unwrap() {
+            match plugin {
+                Ok(path) => match path.file_name().unwrap().to_str().unwrap() {
+                    "requirements.txt" => {
```

Comment:
Ok I think this code is due to some compiler error, I play with it a little bit and you ca use this diff

```diff
diff --git a/reckless_github/src/repository.rs b/reckless_github/src/repository.rs
index e401bcf..8e164d2 100644
--- a/reckless_github/src/repository.rs
+++ b/reckless_github/src/repository.rs
@@ -8,6 +8,7 @@ use reckless_lib::plugin::PluginLang;
 use reckless_lib::repository::Repository;
 use reckless_lib::utils::clone_recursive_fix;
 use reckless_lib::utils::get_dir_path_from_url;
+use reckless_lib::utils::get_plugin_info_from_path;
 
 pub struct Github {
     /// the url of the repository to be able
@@ -41,105 +42,58 @@ impl Github {
         let pattern = format!("{}/[!.]*/*", &repo_path);
         for plugin in glob(&pattern).unwrap() {
             match plugin {
-                Ok(path) => match path.file_name().unwrap().to_str().unwrap() {
-                    "requirements.txt" => {
-                        let path_to_plugin =
-                            String::from(path.parent().unwrap().to_path_buf().to_string_lossy());
-                        let plugin_name = path
-                            .parent()
-                            .unwrap()
-                            .file_name()
-                            .unwrap()
-                            .to_string_lossy()
-                            .to_string();
-                        list_plugins.push(Plugin::new(
-                            &plugin_name,
-                            &path_to_plugin,
-                            PluginLang::Python,
-                        ));
-                    }
-                    "go.mod" => {
-                        let path_to_plugin =
-                            String::from(path.parent().unwrap().to_path_buf().to_string_lossy());
-                        let plugin_name = path
-                            .parent()
-                            .unwrap()
-                            .file_name()
-                            .unwrap()
-                            .to_string_lossy()
-                            .to_string();
-                        list_plugins.push(Plugin::new(
-                            &plugin_name,
-                            &path_to_plugin,
-                            PluginLang::Go,
-                        ));
-                    }
-                    "cargo.toml" => {
-                        let path_to_plugin =
-                            String::from(path.parent().unwrap().to_path_buf().to_string_lossy());
-                        let plugin_name = path
-                            .parent()
-                            .unwrap()
-                            .file_name()
-                            .unwrap()
-                            .to_string_lossy()
-                            .to_string();
-                        list_plugins.push(Plugin::new(
-                            &plugin_name,
-                            &path_to_plugin,
-                            PluginLang::Rust,
-                        ));
-                    }
-                    "pubspec.yaml" => {
-                        let path_to_plugin =
-                            String::from(path.parent().unwrap().to_path_buf().to_string_lossy());
-                        let plugin_name = path
-                            .parent()
-                            .unwrap()
-                            .file_name()
-                            .unwrap()
-                            .to_string_lossy()
-                            .to_string();
-                        list_plugins.push(Plugin::new(
-                            &plugin_name,
-                            &path_to_plugin,
-                            PluginLang::Dart,
-                        ));
-                    }
-                    "package.json" => {
-                        let path_to_plugin =
-                            String::from(path.parent().unwrap().to_path_buf().to_string_lossy());
-                        let plugin_name = path
-                            .parent()
-                            .unwrap()
-                            .file_name()
-                            .unwrap()
-                            .to_string_lossy()
-                            .to_string();
-                        list_plugins.push(Plugin::new(
-                            &plugin_name,
-                            &path_to_plugin,
-                            PluginLang::JavaScript,
-                        ));
-                    }
-                    "tsconfig.json" => {
-                        let path_to_plugin =
-                            String::from(path.parent().unwrap().to_path_buf().to_string_lossy());
-                        let plugin_name = path
-                            .parent()
-                            .unwrap()
-                            .file_name()
-                            .unwrap()
-                            .to_string_lossy()
-                            .to_string();
-                        list_plugins.push(Plugin::new(
-                            &plugin_name,
-                            &path_to_plugin,
-                            PluginLang::TypeScript,
-                        ));
+                Ok(path) => {
+                    let (path_to_plugin, plugin_name) =
+                        get_plugin_info_from_path(path.clone()).unwrap();
+
+                    let file_name = path.file_name().unwrap().to_str().unwrap();
+                    match file_name {
+                        "requirements.txt" => {
+                            list_plugins.push(Plugin::new(
+                                &plugin_name,
+                                &path_to_plugin,
+                                PluginLang::Python,
+                            ));
+                        }
+                        "go.mod" => {
+                            list_plugins.push(Plugin::new(
+                                &plugin_name,
+                                &path_to_plugin,
+                                PluginLang::Go,
+                            ));
+                        }
+                        "cargo.toml" => {
+                            list_plugins.push(Plugin::new(
+                                &plugin_name,
+                                &path_to_plugin,
+                                PluginLang::Rust,
+                            ));
+                        }
+                        "pubspec.yaml" => {
+                            list_plugins.push(Plugin::new(
+                                &plugin_name,
+                                &path_to_plugin,
+                                PluginLang::Dart,
+                            ));
+                        }
+                        "package.json" => {
+                            list_plugins.push(Plugin::new(
+                                &plugin_name,
+                                &path_to_plugin,
+                                PluginLang::JavaScript,
+                            ));
+                        }
+                        "tsconfig.json" => {
+                            // FIXME: avoid to unwrap here
+                            list_plugins.push(Plugin::new(
+                                plugin_name.as_str(),
+                                path_to_plugin.as_str(),
+                                PluginLang::TypeScript,
+                            ));
+                        }
+                        _ => continue,
                     }
-                    _ => continue,
-                },
+                }
                 Err(err) => {
                     RecklessError::new(1, err.to_string().as_str());
                 }

```